sdk/test-utils/recorder/GUIDELINES.md

```diff
@@ -0,0 +1,333 @@
+  const tmr = recorder.newDate("tmr");
+  ```
+
+- When `recorder.stop()` is called for a test, we save that unique information corresponding to the test run along with the test recording in the `record` mode.
+
+- In `playback` mode, the saved unique information is pulled out from the existing recording in order to replay the http requests.
+
+- `newDate: (label: string) => Date;`
+
+  - In live test mode, `new Date()` is returned.
+
+  - In record mode, `new Date()` is returned, and is saved in the recordings by assigning the `label`.
+
+  - In playback mode, the date in the recordings associated to the `label` is returned.
+
+- `getUniqueName: (prefix: string, label?: string) => string`
+
+  - In live test mode, random string is generated, appended to `prefix` and returned.
+
+  - In record mode, random string is generated, appended to `prefix` and returned, and is saved in the test-recordings by assigning the provided `label`.
+
+  - In playback mode, the string in the recordings associated to the `label` is returned.
+
+  - If the `label`(optional param) is not provided, `prefix` is used as the label.
+
+- Any unique information of the test run that is important for playing back the http request must be saved along with the recordings in the record mode.
+
+- If a new test/test-suite is added, execute the test/test-suite(or all the tests) by setting the env variable `TEST_MODE = record` and **commit** the generated recording files.
+
+### Importing `delay` from `@azure/test-utils-recorder`
+
+- This `delay` has no effect if the `TEST_MODE` is `"playback"`.
+- `delay` works as expected(`await delay(<milliseconds>)`) if the `TEST_MODE` is not `"playback"`.
+
+---
+
+## Updating an existing test/test-suite
+
+- Recordings are saved in `./recordings/` folder.
+- If a test is modified, we might need to record it again in order to equip the recording to accommodate any updates in the function calls that invoke http requests.
+
+  - execute the test/test-suite script(or all the tests) by setting the env variable `TEST_MODE = record`
+
+- If only the test title is updated,
+
+  - we can either execute that specific test again to generate the recording
+
+    or
+
+  - we can just update the file name of the recording accordingly
+
+- **Commit** the generated recordings.
+
+---
+
+## Recordings
+
+- Recordings are being saved in one folder for each describe-block test suite
+  recording file structure
+  - `recordings/node/<describe-block-title>/recording_<test-title>.js` for node tests and
+  - `recordings/browsers/<describe-block-title>/recording_<test-title>.json` for browser tests.
+- The file name of the recording for a test preserves the title of the `describe`-block and the corresponding `it`-block (with special characters appropriated). The recordings for node tests go into the `./recordings/node/` folder(Similarly, browser recordings in the `./recordings/browsers/` folder).
+
+  For example
+
+  ```typescript
+  describe("Aborter", () => {
```

Comment:
I thought we weren't using this name

Reply:
Hmm. We haven't updated the title in storage test-suites.

I'll update this to "Some Random Test Suite".